src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/MiningTask.java

```diff
@@ -312,10 +312,17 @@ private int grabFuelFrom(@Nonnull Inventory inv) {
                 ItemStack item = inv.getContents()[i];
 
                 if (fuelType.test(item)) {
-                    ItemUtils.consumeItem(item, false);
-
-                    if (miner instanceof AdvancedIndustrialMiner) {
-                        inv.addItem(new ItemStack(Material.BUCKET));
+                    /*
+                     * Fixes #3336
+                     * Respects the state of the miner if there are
+                     * no any errors during #setPistonState
+                     */
+                    if (running) {
```

Comment:
You can connect these two statements using `&&`, no need to introduce another nesting level here.

Reply:
oopsie my bad, that's a thing I haven't learned from gh where using branches properly, Good to know though. Should I make a new PR with a new branch or keep this one